packages/kit/src/runtime/server/cookie.js

```diff
@@ -26,6 +26,21 @@ function validate_options(options) {
 	}
 }
 
+/**
+ * Generates a unique key for a cookie based on its domain, path, and name in
+ * the format: `<domain>/<path>?<name>`.
+ * If domain is undefined, it will be omitted.
+ * For example: `/?name`, `example.com/foo?name`.
+ *
+ * @param {string | undefined} domain
+ * @param {string} path
+ * @param {string} name
+ * @returns {string}
+ */
+function generate_cookie_key(domain, path, name) {
+	return `${domain || ''}${path}?${name}`;
```

Comment:
The inputs might have to be url encoded to prevent ambiguity. For example:

- `/foo?` `?` `name`, and
- `/foo` `?` `?name`

are currently not distinguishable.

Reply:
yep, now encoding the name